management_api_app/db/repositories/resources.py

```diff
@@ -0,0 +1,26 @@
+from azure.cosmos import CosmosClient
+from pydantic import UUID4
+
+from core import config
+from db.errors import EntityDoesNotExist
+from db.repositories.base import BaseRepository
+from jsonschema import validate
+
+
+class ResourceRepository(BaseRepository):
+    def __init__(self, client: CosmosClient):
+        super().__init__(client, config.STATE_STORE_RESOURCES_CONTAINER)
+
+    @staticmethod
+    def _validate_resource_parameters(workspace_create, workspace_template):
+        validate(instance=workspace_create["properties"], schema=workspace_template)
+
+    def get_resource_dict_by_id(self, workspace_id: UUID4) -> dict:
+        query = self._active_workspaces_query() + f' AND c.id="{workspace_id}"'
```

Comment:
the ResourceRepository class does not appear to have a self._active_workspaces_query() -- and if it should have, it should probably be called _active_resources_query()

the rest of this function also heavily references workspaces.  If all of this is workspace specific perhaps it is better to leave it in the sub class